zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/NettyClientConnectionFactory.java

```diff
@@ -26,42 +24,29 @@
 import io.netty.channel.ChannelInitializer;
 import io.netty.channel.ChannelOption;
 import io.netty.channel.EventLoop;
-import java.net.SocketAddress;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.Objects;
 /**
  * Created by saroskar on 3/16/16.
  */
 public final class NettyClientConnectionFactory {
 
     private final ConnectionPoolConfig connPoolConfig;
     private final ChannelInitializer<? extends Channel> channelInitializer;
-    private final Counter unresolvedDiscoveryHost;
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(NettyClientConnectionFactory.class);
-
 
     NettyClientConnectionFactory(final ConnectionPoolConfig connPoolConfig,
                                  final ChannelInitializer<? extends Channel> channelInitializer) {
         this.connPoolConfig = connPoolConfig;
         this.channelInitializer = channelInitializer;
-        this.unresolvedDiscoveryHost = SpectatorUtils.newCounter("unresolvedDiscoveryHost",
-                connPoolConfig.getOriginName() == null ? "unknownOrigin" : connPoolConfig.getOriginName());
     }
 
     public ChannelFuture connect(final EventLoop eventLoop, SocketAddress socketAddress, CurrentPassport passport) {
+        Objects.requireNonNull(socketAddress, "socketAddress");
         if (socketAddress instanceof InetSocketAddress) {
-            InetSocketAddress inetSocketAddress = (InetSocketAddress) socketAddress;
-            // TODO(cmastrangelo): Make this throw an exception after checking the counter isn't implemented, and do
-            //  the name resolution at a higher level.
-            if (inetSocketAddress.isUnresolved()) {
-                LOGGER.warn("NettyClientConnectionFactory got an unresolved address, addr: {}", inetSocketAddress);
-                unresolvedDiscoveryHost.increment();
-            }
+            // This should be checked by the ClientConnectionManager
+            assert !((InetSocketAddress) socketAddress).isUnresolved() : socketAddress;
```

Comment:
This should ideally live in `PerServerConnectionPool::tryMakingNewConnection` ?

Reply:
It needs to live in CCM because it can return a promise.  PSCP doesn't return a promise, and gets triggered multiple times.    I'd prefer to assert that it is resolved here just before it is actually used, because this method is the one that cares the most about it.   Having the check and the usage very close to each other help understandability.